### PR TITLE
fix(codegen): missing `,` when generating type parameters with jsx

### LINKED ISSUE #7917: bug(codegen): missing `,` when generating type parameters with jsx
[Playground link](https://playground.oxc.rs/#eNo9T7sOwjAM/JXIE0gdAIklPAYGvoCxSwguipTalWMeVZV/J6VqJ5/t851vAA8WPFNS80RCCf5K5mSOt+q8apitua3N6WyGmmrVvkNzcVL2Qz7UlA9QAYMdQF40ltSTui9YlRdWEAPpjJPnDpemb+8c507FUWpYWrCNiwlzBZ2ThFIUCx5VZrxQ/wNQJ08sFoBpt9nuoTA8P7DkmOhtoNCE+biEVOF4jfwZj98od07lp8k05x+N1ltj)

input:

```tsx
const genericFn = <T,>(foo: T) => {
	type Bar = {};
};
```

output:
```tsx
const genericFn = <T>(foo: T) => {
	type Bar = {};
};
```

expected output:
```tsx
const genericFn = <T,>(foo: T) => {
	type Bar = {};
};
```

### PR DESCRIPTION
**Pull request by Dunqing**
close: #7917

### CODE REVIEW

**Comment by Dunqing** [comment]
* **#7929** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7929?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7929?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=7929).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: crates/oxc_codegen/src/gen.rs [hunk 1]
```diff
@@ -2976,6 +2978,8 @@ impl Gen for TSTypeParameterDeclaration<'_> {
             p.print_soft_newline();
             p.dedent();
             p.print_indent();
+        } else if p.is_jsx {
```
Comment (Boshen) [inline crates/oxc_codegen/src/gen.rs:2981]: Can you add a comment here for what source text this is for? i.e.  "<T, >() => {}

Comment (Dunqing) [reply crates/oxc_codegen/src/gen.rs:2981]: Added

**Comment by Boshen** [comment]
### Merge activity

* **Dec 16, 12:33 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

**Pull request merged**